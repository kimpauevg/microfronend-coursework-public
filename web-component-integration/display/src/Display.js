import React, {useState} from 'react';


function Display() {
  const [displayText, setDisplayText] = useState('');

  const listener = new BroadcastChannel('display');

  listener.onmessage = function (message) {
      setDisplayText(displayText + ' ' + message.data);
  }

  const displayBlockStyle=  {
      "background": "#9c61fb",
      "borderRadius": "5px",
      "color": "black",
      "margin": "10px",
      "padding": "10px",
  }

  const displayTextStyle = {
      "border": "1px black solid",
      "borderRadius": "5px",
      "margin": "10px",
      "padding": "10px",
  }

  return (
      <>
          <div style={displayBlockStyle} className="Display-block">
              <h3>This is a remote display component</h3>
              <div style={displayTextStyle} className="Display-text">{displayText}</div>
          </div>
      </>
  );
}

export default Display;
