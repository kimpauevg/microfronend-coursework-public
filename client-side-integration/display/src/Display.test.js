import Display from "./Display";
import renderer from 'react-test-renderer'
import React from 'react';


test('display custom text', () => {
    let component = renderer.create(
        <Display displayText={'test'}></Display>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});