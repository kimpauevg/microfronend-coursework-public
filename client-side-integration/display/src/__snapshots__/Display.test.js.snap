// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display custom text 1`] = `
<div
  className="Display-block"
>
  <h3>
    This is a remote display component
  </h3>
  <div
    className="Display-text"
  >
    test
  </div>
</div>
`;
